index.rst

```diff
@@ -1,60 +1,850 @@
+
+- The next section, :ref:`sec-concepts`, should definitely be read first.
+
+- :ref:`sec-campaign-state` should be read next, though this contains a few forward-references for details (which should mostly be ignored on first read).
+  Expect to go back to this section frequently when reading later sections.
+
+- :ref:`sec-class-and-subcommands` contains numerous references to :ref:`sec-opts` that are often necessary to really understand what a method can and cannot do; readers should probably skim both of these sections before trying to understand either of them in detail.
+  When reading them in detail, it will probably be helpful to have views of both :ref:`sec-opts` and :ref:`sec-class-and-subcommands` open at the same time.
+
+Unfinished Business
+-------------------
+
+While this document represents a fairly complete snapshot of my thinking at one point, some discussions since the last substantive changes have raised some possibilities I'd want to explore further before considering even myself happy with this proposal.
+But it's also past time for me to get this out for review to others, especially considering that there may be other new big ideas out there that should be considered before making any major changes here.
+So I'll just leave most of the content as-is, and list here the big ideas I'd want to explore further:
+
+- Creating a SQLite staging registry (even for local execution), by transferring input datasets needed from an "upstream" shared registry after `~lsst.pipe.base.QuantumGraph` generation.
+  Datasets would only be pushed back to the upstream repo if explicitly requested.
+  This could be a one-step way to provide a lot more insulation between users, and hence make the shared repo access model easier to implement (maybe the only non-admin write operations we need to support are ingest and import?) while also giving users more power to play in their own sandboxes.
+  It also opens up a lot of intriguing possibilities for campaign features (more on that in later bullets).
+  I think the big complications this raises are about how to handle multiple transfers from upstream (if needed), when the staging repo doesn't start out empty and there's potential for conflicts, especially when `~lsst.pipe.base.QuantumGraph` generation is run multiple times over the life of the campaign.
+
+- I don't think I've tried quite hard enough to remove functionality here that isn't obviously needed in the name of simplification.
+  One concrete example is that I don't think anyone really needs to be able to change the input collections after a campaign is created, though it's also impossible to rigorously guard against the possibility that input collection *content* may have changed (unless we only care about changes to a local staging repo!), so that may not be as much of a simplification as I hope.
+
+- We could at least sometimes initialize a git repository in the campaign directory and commit every time we run ``pipetask``.
```

Comment:
This, counter the comment above, does make the `git` analogy more complete, with a local "sandbox" being somewhat similar to a `clone` and with "pushing" that sandbox to a joint permanent repo.